src/device/util.jl

```diff
@@ -132,3 +132,7 @@ function escape_llvm_string(io, s::AbstractString, esc::AbstractString)
     end
 end
 escape_llvm_string(s::AbstractString) = sprint(endof(s), escape_llvm_string, s, "\"")
+
+
+Base.@pure alignment(::Type{T}) where {T} =
+    convert(Int, ccall(:jl_alignment, Cint, (Csize_t,), sizeof(T)))
```

Comment:
You might want to use `jl_datatype_alignment` here or ask LLVM for its preferred alignment. `jl_alignment` is more about GC memory alignment guarantees.